From 65ddd89a18294fea368afd1f41f74396954ce7b0 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 10 Nov 2022 21:54:24 +0000
Subject: [PATCH] DesktopNativeWidgetAura: donc apply DIP for SetBounds

Also, donc scale Display bounds, only its content
should be scaled
---
 src/ui/display/display.cc                                      | 2 +-
 src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ui/display/display.cc b/src/ui/display/display.cc
index 96df627f10..2b40755835 100644
--- a/src/ui/display/display.cc
+++ b/src/ui/display/display.cc
@@ -317,7 +317,7 @@ void Display::SetScaleAndBounds(float device_scale_factor,
   device_scale_factor_ = std::max(0.5f, device_scale_factor_);
 
   gfx::RectF f(bounds_in_pixel);
-  f.Scale(1.f / device_scale_factor_);
+  //f.Scale(1.f / device_scale_factor_);
   bounds_ = gfx::ToEnclosedRectIgnoringError(f, kDisplaySizeAllowanceEpsilon);
   size_in_pixels_ = bounds_in_pixel.size();
   UpdateWorkAreaFromInsets(insets);
diff --git a/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc b/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
index 2825381576..6bf39b7206 100644
--- a/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
+++ b/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
@@ -795,7 +795,7 @@ std::string DesktopNativeWidgetAura::GetWorkspace() const {
 void DesktopNativeWidgetAura::SetBounds(const gfx::Rect& bounds) {
   if (!content_window_)
     return;
-  desktop_window_tree_host_->SetBoundsInDIP(bounds);
+  desktop_window_tree_host_->AsWindowTreeHost()->SetBoundsInPixels(bounds);
 }
 
 void DesktopNativeWidgetAura::SetBoundsConstrained(const gfx::Rect& bounds) {
